client/src/app/+videos/+video-watch/video-watch-playlist.component.ts

```diff
@@ -117,6 +117,7 @@ export class VideoWatchPlaylistComponent {
 
   updatePlaylistIndex (position: number) {
     if (this.playlistElements.length === 0 || !position) return
+    if (position < 0) position = (this.playlistPagination.totalItems + 1) + position // Handle the "reverse" index
```

Comment:
I don't think you need to calculate the position here since you already built it in the video watch component

Reply:
Actually, the position is not properly calculated in the video watch component, because this.playlist is undefined at that time. In the commit I just pushed, I got rid of it. In the video watch component, `this.playlistPosition` can be negative, while `updatePlaylistIndex()` does the job of translating the actually negative position into an index it can use.